docs/access.md

```diff
@@ -0,0 +1,367 @@
+group.
+
+The role data is stored in a config entity type named `OgRole`, and whenever a
+new group type is created, a number of roles will automatically be created:
+
+- `member` and `non-member`: these two roles are required for every group, they
+  indicate whether or not a user is a member.
+- `administrator`: this role is not strictly required but is created by default
+  by OG because it is considered to be generally useful for most groups.
+- Developers can choose to provide additional default roles by listening to the
+  `og.default_role` event.
+
+Whenever a default role is created, it will automatically inherit the
+permissions that are assigned to the role in their permission declaration (see
+the `default role` property above which takes an array of roles to which these
+permissions should be applied).
+
+To manually assign a permission to a role for a certain group, it can be done as
+follows:
+
+```php
+// OgRole::loadByGroupAndName() is the easiest way to load a specific role for
+// a given group.
+$admin_role = OgRole::loadByGroupAndName($this->group, OgRoleInterface::ADMINISTRATOR);
+$admin_role->grantPermission('my permission')->save();
+````
+
+Similarly, an existing permission can be removed from a role by calling
+`$admin_role->revokePermission('my permission')` and saving the role entity.
+
+For more information on how `OgRole` objects are handled, check the following
+methods:
+- `\Drupal\og\GroupTypeManager::addGroup()`: this code is responsible for
+  creating a new group type and will create the roles as part of it.
+- `\Drupal\og\OgRoleManager::getDefaultRoles()`: creates the default roles and
+  fires the event listener that modules can hook into the provide additional
+  default roles.
+- `\Drupal\og\EventSubscriber\OgEventSubscriber::provideDefaultRoles()`: OG's
+  own implementation of the event listener, which provides the `administrator`
+  role.
+
+
+Checking if a user has a certain permission in a group
+------------------------------------------------------
+
+It would be natural to think that checking a permission would be a simple matter
+of loading the `OgRole` entity and calling `$role->hasPermission()` on it, but
+this is not sufficient. There are a number of additional things to consider:
+
+- The super user (user ID 1) has full permissions.
+- OG has a configuration option to allow full access to group owners.
+- Users that have the global permission `administer organic groups` have all
+  permissions in all groups.
+- The role can have the `is_admin` flag set which will grant all permissions.
+- Modules can alter permissions depending on their own requirements.
+
+Organic Groups provides a service that will perform these checks. To determine
+whether the currently logged in user has for example the `manage members`
+permission on a given group entity:
+
+```php
+// Load some custom group.
+$group = \Drupal::entityTypeManager()->getStorage('my_group_type')->load($some_id);
```

Comment:
Yeah it is just an entity type, I didn't want to put `node` here or any other existing entity type since we are entity type agnostic. I agree that we can use a similar document to explain how to programmatically work with group types and group content types.